core/src/avm2/regexp.rs

```diff
@@ -241,21 +241,22 @@ impl<'gc> RegExp<'gc> {
     /// Implements string.replace(regex, replacement) where the replacement may be
     /// a string with $-sequences.
     pub fn replace_string(
-        &mut self,
+        regexp: &Object<'gc>,
```

Comment:
Let's pass `Object` by value, it's just a GC pointer.
(`RegExpObject` with the suggestion above)

Reply:
So I changed this up to `RegExpObject`, but I still have to get to the underlying data  with `regexp.as_regexp_mut(activation.gc()).unwrap()` yeah? I'm not missing some obvious `impl` that can lean on knowing it's a `RegExpObject` a bit more?

Obviously having the signature be more precise here is nicer overall though.

Reply:
Yeah. We could add a new method that returns a non-Option (in fact, I'd claim `as_regexp_mut` being defined on `Object` is a bit of an anti-pattern), but there isn't right now. (and it's not critical to add in this PR specifically)